tests/end_to_end/test_account.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+import pytest_asyncio
+
+from kakarot_scripts.utils.starknet import call
+
+TOTAL_SUPPLY = 10000 * 10**18
+TEST_AMOUNT = 10 * 10**18
+
+
+@pytest.fixture(scope="session")
+async def class_hashes():
+    """
+    All declared class hashes.
+    """
+    from kakarot_scripts.utils.starknet import get_declarations
+
+    return get_declarations()
+
+
+@pytest_asyncio.fixture(scope="function")
+async def token_a(
+    deploy_solidity_contract,
+    owner,
+):
+    return await deploy_solidity_contract(
+        "UniswapV2",
+        "ERC20",
+        TOTAL_SUPPLY,
+        caller_eoa=owner.starknet_contract,
+    )
+
+
+@pytest.mark.asyncio(scope="session")
+@pytest.mark.AccountContract
+class TestAccount:
+    class TestAutoUpgradeOnTransaction:
+        async def test_should_upgrade_outdated_account_on_transfer(
+            self, invoke, token_a, owner, other, class_hashes
+        ):
+            # Given an initial version of 1000
+            prev_version = (await call(owner.starknet_contract.address, "version"))[0]
+            assert prev_version == 1000  # 000.001.000
```

Comment:
I think that you don't need to test against hard coded values

as there is no way to make an external library call, let's just use class hashes?

1. get current class hash
2. assert != class_hashes["account_contract_fixture"]
3. update kakarot
4. make token tx
5. get new class hash
6. assert new class hash == class_hashes["account_contract_fixture"]

?

Reply:
ok, but i didn't understand what you mean by

> as there is no way to make an external library call, let's just use class hashes?

Reply:
I meant that it not possible to get the version of the new and prev class hash with a RPC call